簡単に起動できるようなプルリクを作ってみました

tin-machine opened the pull request:
google-home-voicetext、日常の中で使わせて頂いてます:bow:

ただ一度、サーバの停止などで止まってしまうと再度実行するまでの手順を忘れてしまい
再稼働まで時間がかかってたので自動的にできるように修正してみました。

修正個所などは、それぞれのコミットログの方に記載してみました。

Commit:
トップディレクトリにDockerfileを移動

Commit:
コンテナ起動で自動的に各プロセスが起動するように

ソースコードのcloneをDockerfileであらかじめ行なうような修正です。
docker-compose.yml に

```
    volumes:
      - ./myapps/:/google-home-voicetext/
```

のようにマウントせず、コメントアウトすると、ソースコードのcloneの手間が省けます。

ソース自体の修正を行う場合は、volumesで指定し、 docker-compose up することで、
既存のDockerfilesでcloneしたものを上書きする形でマウントするので、修正できそうです。

RUN mkdir firebase-secret <- この処理ですが、firebaseの鍵を docker-compose.yml の volumes でマウントすると、コンテナ内に入る作業が不要になりました。

起動処理を boot.bash の形でまとめました。

Commit:
起動処理を boot.bash にまとめる

当初の処理からの変更点として

avahi-daemon --daemonize --no-drop-root

Debian環境、Docker環境が悪いのか分からないのですが、「 --no-drop-root 」とroot権限を残した状態でないとavahi-daemonが起動しない事がありました。
--no-drop-root ですが、

```
--no-drop-root
Don't drop root privileges after startup and don't require daemon to be started as root. We recommend not to use this option.
```

参考 https://www.systutorials.com/docs/linux/man/8-avahi-daemon/

また、まれに avahi-daemon の起動に時間がかかるためか、firestore.js, api-server.jsの起動に失敗することがあったのでsleepを入れています。

Commit:
docker-compose.yml変更

1. container_nameをGithubのレポジトリ名に変更
2. working_dir を レポジトリ名に変更( Dockerfileで git clone している関係ですが、 Dockerfileの git clone時にディレクトリ名を指定、変更することもできます )
3. volumes でのマウントポイントの変更

volumesでmyappsをマウントしている個所ですが、
あらかじめDockerfileでgit cloneするように変更できれば、ここはコメントアウトしても動作します。

開発時はvolumesを有効にすれば、Dockerfileでのgit cloneは上書きされます。
ただ開発時の npm install はコンテナ内に入ってから行う必要がありました。

docker-compose run node-google-home /bin/bash
上記で一旦、コンテナに入ってから npm install すると起動しました。

sikkimtemi commented:
@tin-machine ありがとうございます！返信が遅くなって申し訳ありません。

docker-compose.yml変更の修正内容についてですが、環境変数のVOICETEXT_API_KEYをGitHubに載せるのは抵抗があります。
env_file設定を使って環境変数を外部化出来ないでしょうか？
https://matsuand.github.io/docs.docker.jp.onthefly/compose/environment-variables/

外部化した環境変数ファイルは.gitignoreに登録しておけばAPIキーを公開してしまう事故を防げると思います。
いかがでしょうか？

Commit:
docker-composeの環境変数をenv_fileで指定するファイルに移動

https://github.com/sikkimtemi/google-home-voicetext-docker/pull/1#issuecomment-669798707
に対する修正です。

サンプルのファイル名ですが仮にdocker-compose.ymlに指定する環境設定用のファイルは.my.env
.gitignoreで無視するファイル名は *.env としてみました。
*.envと複数の環境変数用のファイルがある前提としたのは、GoogleHomeを複数台あるような環境で、
google-home-voicetextのコンテナを複数台立てたい時に便利かな、と思って複数にしてみました:bow:

tin-machine commented:
たしかに！ありがとうございます、その通りです！
docker-compose.ymlの修正、.gitignoreの修正を行ないました。御都合の良い時に確認をお願いします:bow:

sikkimtemi commented:
@tin-machine 修正ありがとうございます！
マージいたします。
後ほど時間のあるときにREADME.mdも修正しておきます。

tin-machine commented:
@sikkimtemi マージありがとうございます！
README.mdの件すみません、御都合の良いときにでもお願いします :bow:
